shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1079 @@
+            )
+        if picking.state != "assigned":
+            return self._response_for_select_document(
+                message=self.msg_store.stock_picking_not_available(picking)
+            )
+        return self._response_for_select_line(picking)
+
+    def _response_for_select_line(self, picking, message=None):
+        picking.user_id = self.env.user
+        data = {"picking": self._data_for_stock_picking(picking, with_lines=True)}
+        return self._response(next_state="select_line", data=data, message=message)
+
+    def _response_for_confirm_done(self, picking, message=None):
+        picking.user_id = self.env.user
+        data = {"picking": self._data_for_stock_picking(picking, with_lines=True)}
+        return self._response(next_state="confirm_done", data=data, message=message)
+
+    def _response_for_confirm_new_package(
+        self, picking, line, new_package_name, message=None
+    ):
+        picking.user_id = self.env.user
+        data = {
+            "selected_move_line": self._data_for_move_lines(line),
+            "picking": self._data_for_stock_picking(picking, with_lines=True),
+            "new_package_name": new_package_name,
+        }
+        return self._response(
+            next_state="confirm_new_package", data=data, message=message
+        )
+
+    def _select_document_from_product(self, product):
+        """Select the document by product
+
+        next states:
+            - set_lot: a single picking has been found for this packaging
+            - select_document: A single or no pickings has been found for this packaging
+        """
+        move_lines = self._move_line_by_product(product)
+        pickings = move_lines.mapped("move_id.picking_id")
+        if len(pickings) == 1:
+            pickings.user_id = self.env.user
+            if product.tracking not in ("lot", "serial"):
+                return self._response_for_set_quantity(pickings, move_lines)
+            return self._response_for_set_lot(pickings, move_lines)
+        elif len(pickings) > 1:
+            return self._response_for_select_document(
+                pickings=pickings,
+                message=self.msg_store.multiple_picks_found_select_manually(),
+            )
+        # If no available picking with the right state has been found,
+        # return an error
+        return self._response_for_select_document(
+            message=self.msg_store.product_not_found_in_pickings()
+        )
+
+    def _select_document_from_packaging(self, packaging):
+        """Select the document by packaging
+
+        next states:
+            - set_lot: a single picking has been found for this packaging
+            - select_document: A single or no pickings has been found for this packaging
+        """
+        move_lines = self._move_line_by_packaging(packaging)
+        pickings = move_lines.mapped("move_id.picking_id")
+        if len(pickings) == 1:
+            pickings.user_id = self.env.user
```

Comment:
```suggestion
            self._assign_user_to_picking(picking)
```
you know where to use it ;)